pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractDispatcherSingleActiveConsumer.java

```diff
@@ -81,20 +82,33 @@ public AbstractDispatcherSingleActiveConsumer(SubType subscriptionType, int part
 
     protected abstract void cancelPendingRead();
 
-    protected void pickAndScheduleActiveConsumer() {
+    protected void notifyConsumerGroupChanged(Consumer activeConsumer) {
+        if (null != activeConsumer && subscriptionType == SubType.Failover) {
+            consumers.forEach(consumer ->
```

Comment:
We should try to avoid to send multiple "inactive" notifications to a consumer. 
If I was already "inactive", I don't need to be notified when a new consumer is promoted and I'm not involved.

Reply:
I think this is the first iteration for this listener. I am trying to be avoiding any optimization at the beginning and deliver notifications for every changes. This way we can make sure the first version is "eventually" correct without hitting any race condition on handling this kind of smart logic. If sending multiple notifications become a problem we can do a subsequent change later.

does that make sense to you?